addons/hr_expense/models/res_company.py

```diff
@@ -23,5 +23,5 @@ class ResCompany(models.Model):
         "account.payment.method.line",
         string="Payment methods available for expenses paid by company",
         check_company=True,
-        domain="[('payment_type', '=', 'outbound'), ('journal_id', '!=', False)]",
+        domain="[('payment_type', '=', 'outbound'), ('journal_id', '!=', False), ('company_id', '=', company_id)]",
```

Comment:
There is already `check_company=True` here, but it is missing on `<res.config.settings>.company_expense_allowed_payment_method_line_ids`, which is the real issue.

This is needed since payment methods can be shared between companies.

Reply:
Note that the same should be done for `expense_product_id`

Reply:
done !

Reply:
Can you also remove the explicit domain on `company_id` for `expense_product_id`?
It is wrong because it can also be a `parent_of` instead of a `=`.

As a rule of thumb, we should almost never use a domain using explicit clauses for `company_id` (except using `_check_company_domain`)

Reply:
done !